compiler/crates/relay-lsp/src/shutdown.rs

```diff
@@ -16,7 +16,7 @@ pub(crate) fn on_shutdown(
     _state: &impl GlobalState,
     _params: <Shutdown as Request>::Params,
 ) -> LSPRuntimeResult<<Shutdown as Request>::Result> {
-    std::process::exit(0);
+    LSPRuntimeResult::Ok(())
```

Comment:
@captbaritone this might break things internally?

Reply:
Can you explain why this is needed? Since the LSP is supposed to run forever, I think it makes idomatic sense to have it always return an error exit code, but I'm open to learning why we might not want to do that.